src/batch/src/monitor/stats.rs

```diff
@@ -15,147 +15,41 @@
-    pub task_scheduled_duration: LabelGuardedGaugeVec<3>,
-    pub task_slow_poll_duration: LabelGuardedGaugeVec<3>,
-    pub task_mem_usage: LabelGuardedIntGaugeVec<3>,
-}
-
-pub static GLOBAL_BATCH_TASK_METRICS: LazyLock<BatchTaskMetrics> =
-    LazyLock::new(|| BatchTaskMetrics::new(&GLOBAL_METRICS_REGISTRY));
-
-impl BatchTaskMetrics {
-    /// The created [`BatchTaskMetrics`] is already registered to the `registry`.
-    fn new(registry: &Registry) -> Self {
-        let task_labels = ["query_id", "stage_id", "task_id"];
-
-        let task_first_poll_delay = register_guarded_gauge_vec_with_registry!(
-            "batch_task_first_poll_delay",
-            "The total duration (s) elapsed between the instant tasks are instrumented, and the instant they are first polled.",
-            &task_labels,
-         registry).unwrap();
-
-        let task_fast_poll_duration = register_guarded_gauge_vec_with_registry!(
-            "batch_task_fast_poll_duration",
-            "The total duration (s) of fast polls.",
-            &task_labels,
-            registry
-        )
-        .unwrap();
-
-        let task_idle_duration = register_guarded_gauge_vec_with_registry!(
-            "batch_task_idle_duration",
-            "The total duration (s) that tasks idled.",
-            &task_labels,
-            registry
-        )
-        .unwrap();
-
-        let task_poll_duration = register_guarded_gauge_vec_with_registry!(
-            "batch_task_poll_duration",
-            "The total duration (s) elapsed during polls.",
-            &task_labels,
-            registry,
-        )
-        .unwrap();
-
-        let task_scheduled_duration = register_guarded_gauge_vec_with_registry!(
-            "batch_task_scheduled_duration",
-            "The total duration (s) that tasks spent waiting to be polled after awakening.",
-            &task_labels,
-            registry,
-        )
-        .unwrap();
-
-        let task_slow_poll_duration = register_guarded_gauge_vec_with_registry!(
-            "batch_task_slow_poll_duration",
-            "The total duration (s) of slow polls.",
-            &task_labels,
-            registry,
-        )
-        .unwrap();
-
-        let task_mem_usage = register_guarded_int_gauge_vec_with_registry!(
-            "batch_task_mem_usage",
-            "Memory usage of batch tasks in bytes.",
-            &task_labels,
-            registry,
-        )
-        .unwrap();
-
-        Self {
-            task_first_poll_delay,
-            task_fast_poll_duration,
-            task_idle_duration,
-            task_poll_duration,
-            task_scheduled_duration,
-            task_slow_poll_duration,
-            task_mem_usage,
-        }
-    }
-
-    /// Create a new `BatchTaskMetrics` instance used in tests or other places.
-    pub fn for_test() -> Self {
-        GLOBAL_BATCH_TASK_METRICS.clone()
-    }
-}
-
 #[derive(Clone)]
 pub struct BatchExecutorMetrics {
```

Comment:
Maybe rename to `BatchMetrics` or add a comment, since it's not executor-level any more

Reply:
Let's keep using `BatchExecutorMetrics`, but add a comment.